File: tests/e2e/fixtures/deno-hello/main_test.ts
import { greeting } from "./main.ts";

// Zero-import test — avoids `jsr:` / `https:` fetches so `deno test`
// succeeds offline. The monad e2e harness runs this on dev machines
// that may not have network access to jsr.io.
Deno.test("greeting is friendly", () => {
  const got = greeting("world");
  const want = "hello, world";
  if (got !== want) {
    throw new Error(`greeting("world") = ${got}, want ${want}`);
  }
});

File: tests/e2e/fixtures/go-hello/main_test.go
package main

import "testing"

func TestGreeting(t *testing.T) {
	got := Greeting("world")
	want := "hello, world"
	if got != want {
		t.Fatalf("Greeting(%q) = %q, want %q", "world", got, want)
	}
}
